portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,12 +50,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         PreparedQuery results = datastore.prepare(query);
 
         // Create ArrayList of comments and populate from the database
-        ArrayList<String> commentsList = new ArrayList<String>();
+        ArrayList<Comment> commentsList = new ArrayList<Comment>();
         Iterator resultIterator = results.asIterable().iterator();
         for (int i = 0; i < Integer.parseInt(request.getParameter(NUM_COMMENTS_PARAMETER)); i++) {
             if(resultIterator.hasNext()) {
                 Entity commentEntity = (Entity) resultIterator.next();
-                commentsList.add((String) commentEntity.getProperty(TEXT_PROPERTY));
+                try {
+                    commentEntity = datastore.get(commentEntity.getKey());
```

Comment:
Since this also doesn't end up giving strong consistency, can you simply set the `ReadOptions` to `STRONG` and do just make 1 query instead of two. That will also help get rid of the need for try-catch.

Reply:
sounds good.